Add a pre-commit and lint checks for Python

rlmanrique opened the pull request:
Add the possibility to install pre-commits and linters for Python

jfrancoa reviewed (commented):
It would be nice adding some entry on the readme to explain how to install the pre-commit. Or will it work out of the box?

rlmanrique commented:
> It would be nice adding some entry on the readme to explain how to install the pre-commit. Or will it work out of the box?

Yes, you are right. I added an entry in the README :)

rlmanrique committed:
linter: Add pre-commit and linters for Python

Add the possibility to install pre-commits and linters for Python
Together with GitHub action to run the linters

Signed-off-by: Rodrigo Lopez <rodrigo.lopez@weaviate.io>

rlmanrique commented:
I think this is ready now. The change include the linter validation at GitHub action.
